pkg/resource/deploy/source_query.go

```diff
@@ -168,15 +198,61 @@ func runLangPlugin(src *querySource) result.Result {
 
 // newQueryResourceMonitor creates a new resource monitor RPC server intended to be used in Pulumi's
 // "query mode".
-func newQueryResourceMonitor(builtins *builtinProvider, tracingSpan opentracing.Span) (*queryResmon, error) {
+func newQueryResourceMonitor(
+	builtins *builtinProvider, defaultProviderVersions map[tokens.Package]*semver.Version,
+	provs ProviderSource, reg *providers.Registry, plugctx *plugin.Context,
+	providerRegErrChan chan<- result.Result, tracingSpan opentracing.Span) (*queryResmon, error) {
 
 	// Create our cancellation channel.
 	cancel := make(chan bool)
 
+	// Create channel for handling registrations.
+	regChan := make(chan *registerResourceEvent)
+
+	// Create a new default provider manager.
+	var config *Target
+	d := &defaultProviders{
+		defaultVersions: defaultProviderVersions,
+		providers:       make(map[string]providers.Reference),
+		config:          config,
+		requests:        make(chan defaultProviderRequest),
+		regChan:         regChan,
+		cancel:          cancel,
+	}
+
+	go func() {
+		for {
+			select {
+			case e := <-regChan:
+				urn := resource.NewURN(
+					"query-stack", "query-project", "parent-type", e.goal.Type, e.goal.Name)
```

Comment:
This totally sucks, but I don't _think_ we have much precedent for generating a totally fake stack, fake project, and fake parent for a resource type? It seems worse to change how the `Check` interface handles URNs generally, which I think implies that we actually do want to generate this. But something about it just feels kind of wrong. Would love to hear what you think, @pgavlin.